test/test_poolmanager.py

```diff
@@ -83,6 +83,11 @@ def test_nohost(self, url):
         with pytest.raises(LocationValueError):
             p.connection_from_url(url=url)
 
+    def test_host_with_invalid_idna_labels(self):
+        p = PoolManager(5)
+        with pytest.raises(LocationParseError):
+            p.urlopen("GET", "https://.../")
```

Comment:
I'd like to verify the message on the `LocationParseError` being raised here.

It'd also be nice to have test cases for labels that are too long (ie over 64 characters), for `localhost.`, IPv6 with and without zone IDs. Also technically you can have an unencoded `.` in an IPv6 Zone ID, so we need a case for that.

Reply:
I added a test for `labels that are too long` as you mentioned.
For `localhost.` we have already a [test](https://github.com/urllib3/urllib3/blob/fc89ea40881e78028571b723b233e3a8d7d7af75/test/with_dummyserver/test_https.py#L138) which passes through the newly added code.

For IPv6 tests, I don't know how to test it. With the following test:
```python
    def test_ipv6_host(self):
        p = PoolManager(5)
        p.urlopen("GET", "http://[dead::beef]")
```
I will get `urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='dead::beef', port=80)`. I think I need to mock something because `http://[dead::beef]` passes the validation part and it is valid host from validation point of view. Could you please help me?

Reply:
You are right, testing in `PoolManager` level will require mocking the underlying layers. You can have this test in a different file to single out the parsing validation code, look at test/test_util.py for example.

Reply:
Thanks @hodbn, I moved them to `test_util.py` and added missed tests.

Reply:
Looks great 👍 
Just a small test that @sethmlarson mentioned: "an unencoded . in an IPv6 Zone ID".
Also, I think that splitting parameter values to lines is better for readbility:
```python
@pytest.mark.parametrize(
    "label",
    [
        "localhost.",
        "[dead::beef]",
        "[dead::beef%en5]",
    ],
)
```

Reply:
@hodbn Thanks.
I added `"[dead::beef%en5.]"` for `an unencoded . in an IPv6 Zone ID`, Don't know is it a valid case or not?

Reply:
It should be valid.

(Did a brief look at the RFCs, it seems that [RFC6874](https://tools.ietf.org/html/rfc6874#section-2) states that `ZoneID = 1*( unreserved / pct-encoded )`, and the unreserved charaters are defined in [RFC3986](https://tools.ietf.org/html/rfc3986#section-2.3) to be `unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"`.)